src/heatmaps.ts

```diff
@@ -58,7 +59,24 @@ export class Heatmaps {
     }
 
     public get isEnabled(): boolean {
-        return !!this.instance.config.__preview_heatmaps
+        return (
+            !!this.instance.config.__preview_heatmaps ||
+            !!this._enabledServerSide ||
+            !!this.instance.persistence?.props[HEATMAPS_ENABLED_SERVER_SIDE]
+        )
+    }
+
+    public afterDecideResponse(response: DecideResponse) {
+        const optIn = !!response['heatmaps']
+
+        if (this.instance.persistence) {
+            this.instance.persistence.register({
+                [HEATMAPS_ENABLED_SERVER_SIDE]: optIn,
+            })
+        }
+        // store this in-memory in case persistence is disabled
+        this._enabledServerSide = optIn
+        this.startIfEnabled()
```

Comment:
<attempted-nerd-snipe>was thinking this the last time I changed replay, we do this double storage in lots of places, would be so nice if "disabled persistence" was an in-memory store so we didn't need to track things outside of persistence</attempted-nerd-snipe>

Reply:
Hmmmmmmm it already is a memory store 🤔 Which is confusing... I wonder why we have this duplication...?